examples/step-75/step-75.cc

```diff
@@ -0,0 +1,1502 @@
+  // general runtime parameters, level limits, refine and coarsen fractions, as
+  // well as parameters for cell weighting. The actual intention of these
+  // parameters will be described in the upcoming problem class at their
+  // respecitve location where they are used.
+  struct Parameters
+  {
+    unsigned int  n_cycles;
+    double        tolerance_factor;
+    GMGParameters mg_data;
+
+    unsigned int min_h_level;
+    unsigned int max_h_level;
+    unsigned int min_p_degree;
+    unsigned int max_p_degree;
+    unsigned int max_p_level_difference;
+
+    double refine_fraction;
+    double coarsen_fraction;
+    double p_refine_fraction;
+    double p_coarsen_fraction;
+
+    double weighting_factor;
+    double weighting_exponent;
+  };
+
+
+
+  // Now we will finally declare the main class of this program, which solves
+  // the Laplace equation on subsequently refined function spaces. Its structure
+  // will look familiar as it is similar to the main classes of step-27 and
+  // step-40. There are basically just two additions:
+  // - The SparseMatrix object that would hold the system matrix has been
+  //   replaced by an object of the LaplaceOperator class for the MatrixFree
+  //   formulation.
+  // - An object of parallel::CellWeights, which will help us with load 
+  //   balancing, has been added.
+  template <int dim>
+  class LaplaceProblem
+  {
+  public:
+    LaplaceProblem(const Parameters &parameters);
+
+    void run();
+
+  private:
+    void initialize_grid();
+    void setup_system();
+    void print_diagnostics();
+    void solve_system();
+    void compute_indicators();
+    void adapt_resolution();
+    void output_results(const unsigned int cycle);
+
+    MPI_Comm mpi_communicator;
+
+    const Parameters prm;
+
+    parallel::distributed::Triangulation<dim> triangulation;
+    DoFHandler<dim>                           dof_handler;
+
+    hp::MappingCollection<dim> mapping_collection;
+    hp::FECollection<dim>      fe_collection;
+    hp::QCollection<dim>       quadrature_collection;
+    hp::QCollection<dim - 1>   face_quadrature_collection;
+
+    IndexSet locally_owned_dofs;
+    IndexSet locally_relevant_dofs;
+
+    AffineConstraints<double> constraints;
+
+    LaplaceOperator<dim, double>               laplace_operator;
+    LinearAlgebra::distributed::Vector<double> locally_relevant_solution;
+    LinearAlgebra::distributed::Vector<double> system_rhs;
+
+    std::unique_ptr<FESeries::Legendre<dim>>    legendre;
+    std::unique_ptr<parallel::CellWeights<dim>> cell_weights;
+
+    Vector<float> estimated_error_per_cell;
+    Vector<float> hp_decision_indicators;
+
+    ConditionalOStream pcout;
+    TimerOutput        computing_timer;
+  };
+
+
+
+  // @sect3{The <code>LaplaceProblem</code> class implementation}
+
+  // @sect4{Constructor}
+
+  // The constructor starts with an initializer list that looks similar to the
+  // one of step-40. We again ensure that the Triangulation remains smooth,
```

Comment:
I think we can remove any smoothing flags. At least from MG side, we don't need it.